internal/web/context.go

```diff
@@ -0,0 +1,22 @@
+package web
+
+import (
+	"github.com/labstack/echo/v4"
+	"go.uber.org/zap"
+)
+
+type AppContext struct {
+	echo.Context
+	AppLogger *zap.Logger
+}
+
```

Comment:
In the `CreateAppContext` function, consider handling the case where `logger` is nil by either returning an error or initializing with a default logger. This avoids potential runtime panics if `logger` is nil in any future use cases.